gammapy/datasets/simulate.py

```diff
@@ -295,8 +295,12 @@ def event_list_meta(dataset, observation):
         meta["CONV_DEC"] = 0
 
         for idx, model in enumerate(dataset.models):
-            meta["MID{:05d}".format(idx + 1)] = idx + 1
-            meta["MMN{:05d}".format(idx + 1)] = model.name
+            if model.name != dataset.background_model.name:
```

Comment:
Written this way, this can create an empty model ID, if e.g. the background model is second in the list and another source model is defined in third position.
I suppose one could first reorder models and put the background model first and then build the meta table.

Reply:
Thanks  a lot @registerrier . 
I suppose that "reorder the model first and then simulate" can give some troubles to the users, as they have to know that this should be done. However, the tutorial does not currently provide such an information. In addition, doesn't seem strange that MC_ID=0 in the bkg table is hardcoded while, in the metadata, the setting of the correct MC_ID is left to the user? I would suggest to be coherent and make it all hard-coded. 
What about the following solution?

```
idx = 1
for model in dataset.models:
            if model.name != dataset.background_model.name:
                meta["MID{:05d}".format(idx)] = idx
                meta["MMN{:05d}".format(idx)] = model.name
                idx += 1
            else:
                meta["MID{:05d}".format(0)] = 0
                meta["MMN{:05d}".format(0)] = model.name
```

In the case of no background model, the `MapDatasetEventSampler` always crashes. So, as far as I know, we cannot have a background map and no background model.